### PR TITLE
Refactoring: Create common component for social media icons

### PR DESCRIPTION
**Pull request by tejasberi**
- Created common component
- refactored member-ietms

### CODE REVIEW

File: components/member-list-item/index.js [hunk 1]
```diff
@@ -34,43 +32,16 @@ const PreviewMember = (props) => {
...
           {data['twitter_id'] && (
-            <a
-              onClick={onClick}
-              className={classNames.iconContainer}
-              target="_blank"
-              rel="noreferrer"
-              href={`//twitter.com/${data['twitter_id']}`}>
-              <img className={classNames.icon} alt="twitter icon" src="/icons/twitter.png" />
-            </a>
+            <SocialMediaIcon id={data.twitter_id} type="twitter"></SocialMediaIcon>
```
Comment (Shihab-Duzno) [inline components/member-list-item/index.js:35]: the tag can self close

File: components/member-list-item/index.js [hunk 2]
```diff
@@ -1,19 +1,17 @@
-import { string } from 'prop-types';
-import Link from 'next/link';
 import { getDataURL, getImgURL } from 'helper-functions/urls';
+import { useEffect, useRef } from 'react';
```
Comment (Shihab-Duzno) [inline components/member-list-item/index.js:2]: try to keep lib imports on top, followed by custom and local. just a better practice.

File: components/member-list-item/index.js [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
 const PreviewMember = (props) => {
   const { rdsId } = props;
   const { data } = useFetch(getDataURL(rdsId));
   const imgRef = useRef();
 
-  const onClick = (e) => {
-    e.stopPropagation();
-  };
-
   useEffect(() => {
```
Comment (Shihab-Duzno) [inline components/member-list-item/index.js:15]: are we missing fall back?

File: components/member-list-item/member-list-item.module.scss [hunk 4]
```diff
@@ -1,20 +1,23 @@
 .container {
   flex: 1 250px;
-
   display: flex;
   flex-direction: column;
-
   margin: 20px 20px;
```
Comment (Shihab-Duzno) [inline components/member-list-item/member-list-item.module.scss:5]: `margin:20px` would do

File: components/social-media-icon/index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { IconMapper } from './social-media.constant';
```
Comment (Shihab-Duzno) [inline components/social-media-icon/index.js:1]: follow import sequence

File: components/social-media-icon/index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      target="_blank"
+      rel="noreferrer"
+      tabIndex="0"
+      href={`${IconMapper[type].href}/${[id]}`}>
+      <img className={classNames.image} alt={IconMapper[type].alt} src={IconMapper[type].src} />
+    </a>
+  );
+};
+
+SocialMediaIcon.propTypes = {
```
Comment (Shihab-Duzno) [inline components/social-media-icon/index.js:26]: if props are not required, default it using `defaultProps`

File: components/social-media-icon/social-media.constant.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+export const IconMapper = {
+  twitter: {
+    alt: 'twitter',
+    src: 'icons/twitter.png',
+    href: '//twitter.com/'
```
Comment (Shihab-Duzno) [inline components/social-media-icon/social-media.constant.js:5]: `//`?

File: package-lock.json [hunk 8]
```diff
@@ -8188,7 +8188,6 @@
           "version": "2.0.1",
```
Comment (Shihab-Duzno) [inline package-lock.json:8188]: unless something installed new, package-lock.json should change. Please confirm.

File: components/member-list-item/index.js (hunk 2 above)
Comment (tejasberi) [reply components/member-list-item/index.js:2]: Not necessarily, I am using a VS code pluggin called "Sort imports" which auto sorts imports on every save with industry standards.

File: components/member-list-item/index.js (hunk 3 above)
Comment (tejasberi) [reply components/member-list-item/index.js:15]: nope

File: components/member-list-item/member-list-item.module.scss (hunk 4 above)
Comment (tejasberi) [reply components/member-list-item/member-list-item.module.scss:5]: done

File: components/social-media-icon/index.js (hunk 6 above)
Comment (tejasberi) [reply components/social-media-icon/index.js:26]: yup, i would like to keep the proptypes for better understanding of the props

File: components/social-media-icon/social-media.constant.js (hunk 7 above)
Comment (tejasberi) [reply components/social-media-icon/social-media.constant.js:5]: // used for overriding the local / test environment base URL

File: components/social-media-icon/index.js (hunk 5 above)
Comment (tejasberi) [reply components/social-media-icon/index.js:1]: as mentioned, "sort import" VS code extension takes care of industry standard import sequence

File: components/member-list-item/index.js (hunk 1 above)
Comment (tejasberi) [reply components/member-list-item/index.js:35]: done

File: components/member-list-item/member-list-item.module.scss (hunk 4 above)
Comment (tejasberi) [reply components/member-list-item/member-list-item.module.scss:5]: done

File: package-lock.json (hunk 8 above)
Comment (tejasberi) [reply package-lock.json:8188]: not sure how that changes but i have reverted the changes

File: components/member-list-item/index.js (hunk 2 above)
Comment (shihabus) [reply components/member-list-item/index.js:2]: `import { getDataURL, getImgURL } from 'helper-functions/urls';
import { useEffect, useRef } from 'react';

import Link from 'next/link';  // then ?
import SocialMediaIcon from '../social-media-icon';
import classNames from './member-list-item.module.scss';
import { string } from 'prop-types';   // then ?
import useFetch from 'custom-hooks/useFetch';`.  // then ?

File: components/member-list-item/index.js (hunk 3 above)
Comment (shihabus) [reply components/member-list-item/index.js:15]: what if no `data` and `rsId` is coming?

File: components/social-media-icon/index.js (hunk 5 above)
Comment (shihabus) [reply components/social-media-icon/index.js:1]: still messing the imports, this is NIT. Not a major bug.

[official react code](https://github.com/facebook/react/blob/master/fixtures/flight/src/index.js)

File: components/social-media-icon/index.js (hunk 6 above)
Comment (shihabus) [reply components/social-media-icon/index.js:26]: I was asking for something like this (if these are not required props). So they don't break your code.

SocialMediaIcon.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string
};

SocialMediaIcon.defaultProps = {
  id: '',
  type: ''
};

File: components/social-media-icon/social-media.constant.js (hunk 7 above)
Comment (shihabus) [reply components/social-media-icon/social-media.constant.js:5]: ok

File: components/member-list-item/index.js (hunk 3 above)
Comment (tejasberi) [reply components/member-list-item/index.js:15]: i am not getting what u r saying here. Please explain urself and how do u want me to handle it.

File: components/social-media-icon/index.js (hunk 6 above)
Comment (tejasberi) [reply components/social-media-icon/index.js:26]: done

File: package-lock.json (hunk 8 above)
Comment (shihabus) [reply package-lock.json]: package-lock.json is only supposed to change if you add any additional libs(normal case). So if you haven't done it, then practically there shouldn’t be any change in the package-lock.json.

Comment (tejasberi) [reply package-lock.json]: agreed

**Comment by Harshi7016** [comment]
Test

File: components/member-list-item/index.js (hunk 3 above)
Comment (ankushdharkar) [reply components/member-list-item/index.js:15]: @Shihab-Duzno The contract from the API dictates that these fields will have to be present. If the API is not sending it, then we need to catch that on the backend side

**Pull request merged**